### PR TITLE
[Doc] Add update and delete statement

### PR DESCRIPTION
**Pull request by hellolilyliuyi**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: docs/sql-reference/sql-statements/data-manipulation/DELETE.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 Deletes data rows from a table based on the specified conditions. The table can be a partitioned or non-partitioned table.
 
-For Duplicate Key, Aggregate Key, and Unique Key tables, you can delete data from specified partitions. However, the Primary Key table does not allow you to do so. From v2.3, Primary Key tables support complete `DELETE...WHERE` semantics, which allows you to delete data rows based on the primary key, any column, or the results of a subquery.
+For Duplicate Key, Aggregate Key, and Unique Key tables, you can delete data from specified partitions. However, the Primary Key table does not allow you to do so. From v2.3, Primary Key tables support complete `DELETE...WHERE` semantics, which allows you to delete data rows based on the primary key, any column, or the results of a subquery. From v3.0, StarRocks enriches the `DELETE...WHERE` semantics with multi-table joins and common table expressions (CTEs). If you need to join the Primary Key table with other tables in the database, you can reference these other tables in the USING clause or CTE.
```
Comment (amber-create) [inline docs/sql-reference/sql-statements/data-manipulation/DELETE.md:5]: image
复数，非特指，改为 -> However, Primary Key tables do not allow you to do so.

Comment (amber-create) [inline docs/sql-reference/sql-statements/data-manipulation/DELETE.md:5]: image
非特指

File: docs/sql-reference/sql-statements/data-manipulation/DELETE.md [hunk 2]
```diff
@@ -152,15 +152,20 @@ From v2.3, Primary Key tables support complete `DELETE...WHERE` semantics, which
...
+|   **Parameter**   | **Required** | **Description**                                              |
+| :---------------: | :----------- | :----------------------------------------------------------- |
+|   `with_query`    | No           | One or more CTEs that can be referenced by name in a DELETE statement. CTEs are temporary result sets that can improve the readability of complex statements. |
+|   `table_name`    | Yes          | The table from which you want to delete data.                |
+|    `from_item`    | No           | One or more other tables in the database. These tables can be joined with the table being operated based on the condition specified in the WHERE clause. Based on the result set of the join query, StarRocks deletes the matched rows in the table being operated. For example, if the USING clause is `USING t1 WHERE t0.pk = t1.pk;`, StarRocks converts the table expression in the USING clause to `t0 JOIN t1 ON t0.pk=t1.pk;` when executing the DELETE statement. |
```
Comment (amber-create) [inline docs/sql-reference/sql-statements/data-manipulation/DELETE.md:167]: image
delete xxx from xxx 好一些

File: docs/sql-reference/sql-statements/data-manipulation/DELETE.md [hunk 3]
```diff
@@ -152,15 +152,20 @@ From v2.3, Primary Key tables support complete `DELETE...WHERE` semantics, which
...
+| `where_condition` | Yes          | Only rows that meet the WHERE condition can be deleted. This parameter is required, because it helps prevent you from accidentally deleting the entire table. If you want to delete the entire table, you can use 'WHERE true'. |
```
Comment (amber-create) [inline docs/sql-reference/sql-statements/data-manipulation/DELETE.md:168]: image
文档+信息完整性。前面加一句参数描述 -> The condition based on which you want to delete rows.

File: docs/sql-reference/sql-statements/data-manipulation/UPDATE.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
+# UPDATE
+
+Modifies rows in a Primary Key table. In versions earlier than version 3.0, the UPDATE statement only supports simple syntax, such as `UPDATE <table_name> SET <column_name>=<expression> WHERE <where_condition>`. Starting from version 3.0, StarRocks enriches the syntax, to do multiple table joins and common table expressions (CTEs). If you need to join the table to be updated with other tables in the database, you can reference these other tables in the FROM clause or CTE.
```
Comment (amber-create) [inline docs/sql-reference/sql-statements/data-manipulation/UPDATE.md:3]: image
语法+术语正确性。改为 -> StarRocks enriches the syntax to support multi-table joins and common table expressions (CTEs).

File: docs/sql-reference/sql-statements/data-manipulation/UPDATE.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+The expression that assigns new values to the column.
+
+`from_item`
+
+One or more other tables in the database. These tables can be joined with the table to be updated based on the condition specified in the WHERE clause. The values of the rows in the result set are used to update the values for the specified columns in the matched rows in the table to be updated. For example, if the FROM clause is 'FROM t1 WHERE t0.pk = t1.pk', StarRocks converts the table expression in the FROM clause to 't0 JOIN t1 ON t0.pk=t1.pk' when executing the UPDATE statement.
+
+`where_condition`
+
+Only rows that meet the WHERE condition can be updated. This parameter is required, because it helps prevent you from accidentally updating the entire table. If you want to update the entire table, you can use 'WHERE true'.
```
Comment (amber-create) [inline docs/sql-reference/sql-statements/data-manipulation/UPDATE.md:39]: image
参照前面，也加一句描述

**Review by evelynzhaojie — APPROVED** [review]

**Review by amber-create — APPROVED** [review]

**Comment by hellolilyliuyi** [comment]
@mergify backport branch-3.0

**Pull request merged**